RuleDocumentation/PossibleIncorrectComparisonWithNull.md

```diff
@@ -7,7 +7,7 @@
 To ensure that PowerShell performs comparisons correctly, the `$null` element should be on the left side of the operator.
 
 There are a number of reasons why this should occur:
-* When there is an array on the left side of a null equality comparison, PowerShell will check for a `$null` IN the array rather than if the array is null.
+* When there is an array on the left side of a null equality comparison, PowerShell will create a new array containing one `$null` item for each `$null` in the original array rather than check if the array is null.
```

Comment:
I think `Get-Help  about_Comparison_Operators` explains this much better:

"When the input to an operator is a scalar value, comparison operators return a Boolean value. When the input is a collection of values, the comparison operators return any matching values. If there are no matches in a collection, comparison operators do not return anything."

Effectively by putting `$null` on the left hand side, you enforce `PowerShell` to use the scalar equality comparer, which guarantees a boolean return value (because if `$null` was on the right hand side, it can return nothing, which can lead e.g. to cases where e.g. `if($obj -eq $null)` always goes through the `if` path). I think it would be better to give examples where stuff can go wrong like e.g.:
`if( (@() -eq $null) -and (@() -ne $null) ){"This is logically impossible but correct in PowerShell"}`